### PR TITLE
Fixes #4037. IN operator contains space and `\n` `\t` `\r`

### LINKED ISSUE #4037: sharding+encrypt when sql contains "\n"，sql rewrite error  for mybatis. （in）
### Which version of ShardingSphere did you use?
4.0.0  and dev branch
### Which project did you use? Sharding-JDBC or Sharding-Proxy?
sharding-jdbc
### Expected behavior
the `(`  are not in pairs , there are 5 `(`  in the actual SQL.
### Actual behavior
```
ShardingSphere-SQL                       : Logic SQL: select
    *
    from user_info
     WHERE (  mobile in
                  (
                    ?
                  ) )
```
```
ShardingSphere-SQL                       : Actual SQL: ds1 ::: select
    *
    from user_info_1
     WHERE (  mobile in
                  (
                   (?) ) ::: [15xxxxxxxxxx]
```

### PR DESCRIPTION
**Pull request by xiyelife**
Fixes #4037.

Changes proposed in this pull request:
- parse the IN operator bracket's index for encrypt rewirte
- fixes encrypt statement IN operator  contains space and `\n` `\t` `\r`

### CODE REVIEW

File: encrypt-core/encrypt-core-rewrite/src/test/resources/encrypt/select_for_query_with_cipher.xml [hunk 1]
```diff
@@ -36,4 +36,26 @@
         <input sql="SELECT a.account_id, a.password, a.amount AS a, a.status AS s FROM t_account_bak AS a WHERE a.account_id = 1 AND a.password = 'aaa' AND a.amount = 1000 AND a.status = 'OK'" />
         <output sql="SELECT a.account_id, a.cipher_password, a.cipher_amount AS a, a.status AS s FROM t_account_bak AS a WHERE a.account_id = 1 AND a.assisted_query_password = 'assisted_query_aaa' AND a.cipher_amount = 'encrypt_1000' AND a.status = 'OK'" />
     </rewrite-assertion>
+
+    <rewrite-assertion id="select_plain_for_parameters_with_in_no_space1">
```
Comment (terrymanu) [inline encrypt-core/encrypt-core-rewrite/src/test/resources/encrypt/select_for_query_with_cipher.xml]: please use meaningful name, avoid name as `xxxx1` or `xxxx2`

Comment (xiyelife) [reply encrypt-core/encrypt-core-rewrite/src/test/resources/encrypt/select_for_query_with_cipher.xml]: I see.

**Comment by terrymanu** [comment]
Can you submit a pull request to master as well?

**Comment by xiyelife** [comment]
> Can you submit a pull request to master as well?

I have fixed it by this PR https://github.com/apache/incubator-shardingsphere/pull/4193

**Pull request merged**